specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchservice.json

```diff
@@ -7215,6 +7310,51 @@
       },
       "description": "A skill to split a string into chunks of text."
     },
+    "CustomEntityLookupSkill": {
+      "x-ms-discriminator-value": "#Microsoft.Skills.Text.CustomEntityLookupSkill",
+      "allOf": [
+        {
+          "$ref": "#/definitions/SearchIndexerSkill"
+        }
+      ],
+      "properties": {
+        "defaultLanguageCode": {
+          "$ref": "#/definitions/CustomEntityLookupSkillLanguage",
+          "description": "A value indicating which language code to use. Default is en."
+        },
+        "entitiesDefinitionUri": {
+          "type": "string",
+          "description": "Path to a JSON or CSV file containing all the target text to match against."
+        },
+        "inlineEntitiesDefinition": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CustomEntity"
+          },
+          "description": "The inline CustomEntity definition."
+        },
+        "globalDefaultCaseSensitive": {
+          "type": "boolean",
+          "x-nullable": true,
```

Comment:
Why is this property both optional and nullable? What is the difference between the key/value not being present and the key/value being present with a `null` value?

Reply:
For some reason all my replies seem gone...

The key/value not being present and key/value being present with a null value don't make a difference from the backend service perspective. Do you recommend me to remove "x-nullable" here and for all other similar fields?

Reply:
I would recommend for the service to not send the `null` value on the wire if that is the case. If the service isn't actually sending `null` on the wire, then the `x-nullable` annotation should be removed.

If the service *is* sending a null value on the wire, it is unfortunate, but it has to be marked as x-nullable in the swagger. At the end of the day, the swagger has to describe what how the service is behaving regardless of if it is following best practices or not...

Reply:
I recently had this question with @bleroy and @brjohnstmsft. Not only is it a waste of bandwidth, but it has been causing problems since not nearly everything that should be marked `x-nullable` is, and our generators do care that it's authored correctly. Alas, it sounds like the OData library they use does this. Still, is there some way to get that fixed, or opt into a more wire-friendly serializer? I know Newtonsoft.Json won't send explicit nulls by default, so it seems OData is overriding that. Why, I can't imagine.

Reply:
>  I know Newtonsoft.Json won't send explicit nulls by default, so it seems OData is overriding that. 

OData .NET has its own serializer; It's not based on JSON.NET.

Trying to fix this would be a potentially costly change for us (either swapping out OData's serializer, or forking the library and fixing it), and would be hard to justify given our other priorities. You should proceed on the assumption that the service behaviour isn't going to change.